rdm/0003-persistent-identifiers.md

```diff
@@ -0,0 +1,105 @@
+- Start Date: 2019-09-30
+- RFC PR: [#3](https://github.com/inveniosoftware/rfcs/pull/3)
+- Authors: Lars Holm Nielsen
+
+# Persistent identifiers
+
+## Summary
+
+Overall design for management of persistent identifiers in Invenio RDM.
+
+## Motivation
+
+- As a user, I want to have a DOI for my upload, so that I I can cite my research.
+- As a admin, I want to manage multiple DOI prefixes, so that I I don’t need a repository installation per prefix.
+- As a admin, I want to assign both DOIs and Handles for uploads to my repository, so that I I can fulfill business requirements.
+- As a admin, I want to have a Handle for one upload and a DOI for another upload, so that I I don’t need a repository installation per PID scheme.
+- As a user, I want to share a dataset that already have a DOI, so that I I can collect all my research one place.
+- As a admin, I want to have tombstone pages, so that I I can remove content.
+- As a admin, I want to take over management of a dataset with a DOI from another repository, so that I records can be migrated between repositories..
+- As a admin, I want to customize which internal PID scheme is used, so that I I can continue my existing identifier scheme..
+- As a admin, I want to ensure all records have a DOI, so that I all records have an DOI.
+- As a user, I want to reserve a DOI prior to uploading my report, so that I I can include the DOI in the report.
+
+## Detailed design
+
+### Internal persistent identifier
+
+#### One internal PID per record
+All records must have an internal persistent identifier that is unique for the given data model. Assigning an internal persistent identifier ensures that all records have an identifier from the same identifier scheme. This is useful e.g. for identifier resolution in the REST API as well as for managing relationships internally.
+
+InvenioRDM should further allow administrators to easily configure which internal identifier scheme should be used. The only two identifier schemes provided by default are:
+
+- Record identifier (randomly generated 8 character string)
+- Recid (Invenio v1 style auto-incrementing integers)
+
+#### Internal PID: Record identifier (random number encoded as 8 characters)
+The internal persistent identifier is named the record identifier. The *record identifier* is a randomly generated number encoded using base32. The result is an 8 character string where the last character is a checksum. Following is an example of a record identifier:
```

Comment:
DataCite is using a mod 97-10, ISO 7064 checksum as implemented in https://github.com/datacite/base32-url. I haven't found a Python implementation, but it probably makes sense to be consistent.

Reply:
We have now implemented this in inveniosoftware/base32-lib . mod 97-10 is used there and it appends 2 checksum digits at the end of the string. To have a large record identifier pool with checksums, maybe we should elevate the size to 10 characters (with the last 2 being checksums), effectively giving us (2^5)^8 = 2^40 =~ 10^12 =~ a trilion possible records? Example being:

```
3sbk2-5j060
```